### ISSUE TITLE
Add new metrics and objectives

### ISSUE THREAD

**Issue by annaveronika** [catboost/catboost#203]
Currently we have:
RMSE
Logloss
MAE
CrossEntropy
Quantile
LogLinQuantile
MultiClass
MultiClassOneVsAll
MAPE
Poisson
PairLogit
QueryRMSE
QueryAverage
Recall
Precision
F1
TotalF1
AUC
Accuracy
R2
MCC
PairAccuracy

Is would be cool to add more of them.

**Comment by DictMiner** [comment]
For help: https://www.kaggle.com/wiki/Metrics
esp. GINI --> Normalized GINI

**Comment by souravsingh** [comment]
@annaveronika I would be interested in adding Hamming Loss. Where do I start?

**Comment by annaveronika** [comment]
Great!

You need to add a class here:
https://github.com/catboost/catboost/blob/master/catboost/libs/metrics/metric.cpp

Add a enum value.

Then add a case in CreateMetric here:
https://github.com/catboost/catboost/blob/master/catboost/libs/metrics/metric.cpp#L1125

And also write text that we will add to documentation here:
https://tech.yandex.com/catboost/doc/dg/concepts/loss-functions-docpage/

**Comment by JoshuaC3** [comment]
Built in Fair Loss function. See https://github.com/catboost/catboost/issues/114 and https://github.com/Microsoft/LightGBM/issues/356

**Comment by sergun** [comment]
@annaveronika do you support multi-output regression task?

**Comment by annaveronika** [comment]
@sergun, not yet, but it would be really nice to add

**Comment by Guillermogsjc** [comment]
hi, it looks like **AUC** evaluation metric with `MultiClass` objective is not working properly, looks like if it would be taking in consideration only two categories in target variable, or it is simply broken.

From this description `This function is calculated separately for each class k numbered from 0 to M – 1.` i think that the `AUC` evaluation metric is the mean of the `AUC` of every softmax predicted column, right?

Version 0.8.1.1 - python package

Thank you very much

**Comment by nikitxskv** [comment]
@Guillermogsjc Hi!
For now you can find desired AUC metric values in the log files that are recorded during training (`learn_error.tsv/test_error.tsv` or `catboost_training.json`).

**Comment by DragonTotemRX** [comment]
@annaveronika hi,

I was trying to use CatBoost regressor to train a model to predict retail sales (poisson distribution). And I put custom_metric, loss_function and eval_metric as 'Poisson'. then I got lots of negative predictions. Would you please share your knowledge with me how to use Poisson regression in CatBoost and if it is possible, is there any working example I can learn on poisson regression using catboost? Please give me advice. Thank you.

**Comment by hahlw** [comment]
how can I use the Huber metrics

**Comment by hahlw** [comment]
now this error is Unknown loss function Huber

**Comment by annaveronika** [comment]
@hahlw Do you use the latest version? It's only available starting from 0.15

**Comment by hahlw** [comment]
OK，thanks for your reply



------------------ Original ------------------

**Comment by parisaazimaee** [comment]
@annaveronika I had the same issue as @DragonTotemRX . I was interested to use count:Poisson objective of XGBoost but the closest option was Poisson objective in Catboost. but I got negative predictions which does not make any sense. I'll appreciate your help.

**Comment by annaveronika** [comment]
> @annaveronika I had the same issue as @DragonTotemRX . I was interested to use count:Poisson objective of XGBoost but the closest option was Poisson objective in Catboost. but I got negative predictions which does not make any sense. I'll appreciate your help.

Sorry for late response, we missed it here and then forgot to answer.
Anyway to get the result from formula value you need to take an exponent of it.
We'll add new prediction type for Poisson and use it in predict method, here is an issue for that: https://github.com/catboost/catboost/issues/1184
But for now please use the exponentiation.
We will also add a tutorial specifically for poisson regression:
https://github.com/catboost/catboost/issues/1185

I also remember that you had some other questions, could you repeat them here or in a new issue?

**Comment by annaveronika** [comment]
Everything that was mentioned here is already implemented, there are other issues for some particular metrics, so I'm closing this issue.
Please create new ones if some useful metric is missing.
